ping/ping.go

```diff
@@ -0,0 +1,212 @@
+				DetermineNextPingTime := SortedPingSlice[i+1]
+				DetermineNextPingTime -= SortedPingSlice[i]
+				if DetermineNextPingTime > Set.MaxNextPing {
+					selected = NodeResults[SortedPingSlice[i]]
+					logging.Infof("%s selected, next node has too high ping time\n", NodeResults[SortedPingSlice[i]])
+					break
+				}
+				logging.Warnf("%s had more than %v miners, trying new inorder to retain efficiency\n", SortedPingSlice[i], Set.MaxMiners)
+			}
+		} else {
+			logging.Warnf("%s had more than a %f fee, trying new\n", SortedPingSlice[i], Set.MaxFee)
+		}
+	}
+	return selected
+}
+
+func pingNodes() error {
+	results := make([]time.Duration, len(nodes))
+
+	for i := 0; i < len(nodes); i++ {
+		pinger, err := ping.NewPinger(nodes[i])
+		pinger.SetPrivileged(true)       //This line is needed for windows because of ICMP
+		pinger.Timeout = Set.PingTimeout //Sets the time for which the pinger will timeout regardless of how many packets there has been recieved
+		if err != nil {
+			logging.Warn("Error: Check if you are connected to the internet")
+			logging.Warn(err)
+			return err
+		}
+		pinger.Count = Set.PingPackets //Number of packets to be sent to each node
+		err = pinger.Run()
+		if err != nil {
+			logging.Warn("Error: Check if you are connected to the internet")
+			logging.Warn(err)
+			return err
+		}
+		if nodes[i] == "p2proxy.vertcoin.org" { //Currently this node uses port 9172 instead of 9171, if this changes this statement can be removed and 9171 can be added to all nodes
+			nodes[i] += ":9172"
+		} else {
+			nodes[i] += ":9171"
+		}
+		results[i] = pinger.Statistics().AvgRtt
+		NodeResults[results[i]] = nodes[i]
+		logging.Infof("%s: %v \n", nodes[i], results[i])
+	}
+	return nil
+}
+
+func SortPingtimes() []time.Duration {
+	SortedMap := make([]time.Duration, 0, len(NodeResults))
+	for k := range NodeResults {
+		SortedMap = append(SortedMap, k)
+	}
+	sort.Slice(SortedMap, func(i, j int) bool { return SortedMap[i] < SortedMap[j] })
+	return SortedMap
+}
+
+func getNodeInformation(Node string) (jsonPayload map[string]interface{}, err error) {
+	NodeURL := "http://"
+	NodeURL += Node
+	err = util.GetJson(fmt.Sprintf("%s/local_stats", NodeURL), &jsonPayload)
+	if err != nil {
+		logging.Errorf("Unable to fetch node information\n", err.Error())
+		return jsonPayload, err
+	}
+	return jsonPayload, nil
+}
+
+func CheckFee(jsonPayload map[string]interface{}) bool {
+	fee, ok := jsonPayload["fee"].(float64)
+	if !ok {
+		return false
+	}
+	donationFee, ok := jsonPayload["donation_proportion"].(float64)
+	if !ok {
+		return false
+	}
+	fee += donationFee
+	if fee > Set.MaxFee {
+		return false
+	}
+	return true
+}
+
+//To ensure efficiency of the selected p2pool node a limit of miners has been put in place, returns true if the number is equal to Maxminers or below
+func CheckCurrentMiners(jsonPayload map[string]interface{}) bool {
+	CurrentMiners, ok := jsonPayload["miner_hash_rates"].(string)
```

Comment:
Local variables should not be capitalized. Not sure if this is a rule in Go, but it's not done anywhere else in OCM

Reply:
I don't believe it is but it's good ettique, must've slipped my eye.